pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -247,14 +249,19 @@ func (c *Controller) QueueStatusUpdates(r GatewayResources) {
 }
 
 func (c *Controller) handleStatusUpdates(configs []config.Config) {
-	if c.statusController == nil || !c.statusEnabled.Load() {
+	if c.statusController.Load() == nil || !c.statusEnabled.Load() {
 		return
 	}
 	for _, cfg := range configs {
 		ws := cfg.Status.(*kstatus.WrappedStatus)
 		if ws.Dirty {
 			res := status.ResourceFromModelConfig(cfg)
-			c.statusController.EnqueueStatusUpdateResource(ws.Unwrap(), res)
+			statusController := c.statusController.Load()
+			if statusController == nil {
+				// leader election loss
+				break
+			}
+			statusController.EnqueueStatusUpdateResource(ws.Unwrap(), res)
 		}
 	}
 }
```

Comment:
```suggestion
func (c *Controller) handleStatusUpdates(configs []config.Config) {
	statusController := c.statusController.Load()
	if statusController == nil || !c.statusEnabled.Load() {
		return
	}
	for _, cfg := range configs {
		ws := cfg.Status.(*kstatus.WrappedStatus)
		if ws.Dirty {
			res := status.ResourceFromModelConfig(cfg)
			statusController.EnqueueStatusUpdateResource(ws.Unwrap(), res)
		}
	}
}
``

Reply:
Your code works as well I just don't think there is a need to do the checks multiple times like we had?